bin/oref0-setup.sh

```diff
@@ -1115,17 +1132,30 @@ if [[ $REPLY =~ ^[Yy]$ ]]; then
           wget -qO- $downloadUrl | tar xJv -C $HOME/go/bin || die "Couldn't download and extract Go pump binaries"
           echo "Installing Go pump binaries ..."
           ln -sf $HOME/go/bin/openaps.jq $directory/ || die "Couldn't softlink openaps.jq"
-          rsync -rtuv $HOME/go/bin/ /usr/local/bin/ || die "Couldn't rsync go/bin"
+          cp -pruv $HOME/go/bin/* /usr/local/bin/ || die "Couldn't copy go/bin"
           mv /usr/local/bin/mmtune /usr/local/bin/Go-mmtune || die "Couldn't mv mmtune"
         fi
     fi
     if [[ ${CGM,,} =~ "g4-go" ]]; then
-        if egrep -i "edison" /etc/passwd 2>/dev/null; then
-            go get -u -v -tags nofilter github.com/ecc1/dexcom/...
+        if [ ! -d $HOME/go/bin ]; then mkdir -p $HOME/go/bin; fi
+        echo "Installing or Compiling Go dexcom binaries ..."
+        if $buildgofromsource; then
+            if egrep -i "edison" /etc/passwd &>/dev/null; then
+                go get -u -v -tags nofilter github.com/ecc1/dexcom/...
+            else
+                go get -u -v github.com/ecc1/dexcom/...
+            fi
         else
-            go get -u -v github.com/ecc1/dexcom/...
+            arch=arm
+            if egrep -i "edison" /etc/passwd &>/dev/null; then
+                arch=386
+            fi
+            downloadUrl=$(curl -s https://api.github.com/repos/ecc1/dexcom/releases/latest | \
+            jq --raw-output '.assets[] | select(.name | contains("'$arch'")) | .browser_download_url')
+            echo "Downloading Go dexcom binaries from:" $downloadUrl
+            wget -qO- $downloadUrl | tar xJv -C $HOME/go/bin || die "Couldn't download and extract Go dexcom binaries"
         fi
-        rsync -rtuv $HOME/go/bin/ /usr/local/bin/ || die "Couldn't rsync go/bin"
+        cp -pruv $HOME/go/bin/* /usr/local/bin/ || die "Couldn't copy go/bin"
```

Comment:
What's the advantage of using cp over rsync?

Reply:
consistency, cp is more universal, I had a commit a long while back and we changed other rsync's over to cp